src/Symfony/Component/HttpClient/CurlHttpClient.php

```diff
@@ -34,15 +35,18 @@ final class CurlHttpClient implements HttpClientInterface
 
     private $defaultOptions = self::OPTIONS_DEFAULTS;
     private $multi;
+    private $logger;
 
     /**
      * @param array $defaultOptions     Default requests' options
      * @param int   $maxHostConnections The maximum number of connections to a single host
      *
      * @see HttpClientInterface::OPTIONS_DEFAULTS for available options
      */
-    public function __construct(array $defaultOptions = [], int $maxHostConnections = 6)
+    public function __construct(LoggerInterface $logger = null, array $defaultOptions = [], int $maxHostConnections = 6)
```

Comment:
This should be the last argument IMHO.

Reply:
That does make it harder to pass the default values properly (in case they change) on line 165: `$this->multi->handle = (new self($this->logger))->multi->handle;`

Reply:
True, what about something like `$this->multi->handle = (clone $this)->multi->handle;`, with any proper cloning handling if needed?

Reply:
We would then have to empty the clone

Reply:
What about putting the argument 2nd?
Looks best ordering to me, from most often passed to most often left to defaults.

Reply:
Alternatively: withLogger()

Reply:
Having it last in the constructor and/or using a mutator looks more legit, since from a strict object point of view, if we create it the most needed is to configure it. The logger is not a requirement for this client, this is obviously why we _add_ it in this PR. I don't think we should break the previous implementation. Plus using a logger also surely means in most cases that DI is used and we don't care about having it third.

Reply:
Or it could implement `Psr\Log\LoggerAwareInterface` (adds a `setLogger()`) method. This interface triggers the autoconfiguration in Symfony IIRC.

Reply:
I like @dunglas suggestion.